Formula/v/virtctl.rb

```diff
@@ -0,0 +1,20 @@
+class Virtctl < Formula
+  desc "Allows for using more advanced kubevirt features"
+  homepage "https://github.com/kubevirt/kubevirt"
+  url "https://github.com/kubevirt/kubevirt/archive/refs/tags/v1.1.0.tar.gz"
+  sha256 "3f51390dfcd16de55ba9dd8eac6f587916fdccb979df71003bf26b00de3ee7c4"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = "-X 'kubevirt.io/client-go/version.gitVersion=#{version}'"
+    system "go", "build", *std_go_args(ldflags: ldflags), "./cmd/virtctl"
+
+    generate_completions_from_executable(bin/"virtctl", "completion")
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/virtctl version -c")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.